mmtrack/models/task_modules/track/aflink.py

```diff
@@ -0,0 +1,265 @@
+    `StrongSORT<https://arxiv.org/abs/2202.13514>`_.
+
+    Args:
+        checkpoint (str): Checkpoint path.
+        temporal_threshold (tuple, optional): The temporal constraint
+            for tracklets association. Defaults to (0, 30).
+        spatial_threshold (int, optional): The spatial constraint for
+            tracklets association. Defaults to 75.
+        confidence_threshold (float, optional): The minimum confidence
+            threshold for tracklets association. Defaults to 0.95.
+    """
+
+    def __init__(self,
+                 checkpoint: str,
+                 temporal_threshold: tuple = (0, 30),
+                 spatial_threshold: int = 75,
+                 confidence_threshold: float = 0.95):
+        self.temporal_threshold = temporal_threshold
+        self.spatial_threshold = spatial_threshold
+        self.confidence_threshold = confidence_threshold
+
+        self.model = AFLinkModel()
+        if checkpoint:
+            load_checkpoint(self.model, checkpoint)
+        if torch.cuda.is_available():
+            self.model.cuda()
+        self.model.eval()
+
+        self.device = next(self.model.parameters()).device
+        self.fn_l2 = lambda x, y: np.sqrt(x**2 + y**2)
+
+    def data_transform(self,
+                       track1: np.ndarray,
+                       track2: np.ndarray,
+                       length: int = 30) -> Tuple[Tensor]:
+        """Data Transformation. This is used to standardize the length of
+        tracks to a unified length. Then perform min-max normalization to the
+        motion embeddings.
+
+        Args:
+            track1 (ndarray): the first track with shape (N,C).
+            track2 (ndarray): the second track with shape (M,C).
+            length (int): the unified length of tracks. Defaults to 30.
+
+        Returns:
+            Tuple[Tensor]: the transformed track1 and track2.
+        """
+        # fill or cut track1
+        length_1 = track1.shape[0]
+        track1 = track1[-length:] if length_1 >= length else \
+            np.pad(track1, ((length - length_1, 0), (0, 0)))
+
+        # fill or cut track1
+        length_2 = track2.shape[0]
+        track2 = track2[:length] if length_2 >= length else \
+            np.pad(track2, ((0, length - length_2), (0, 0)))
+
+        # min-max normalization
+        min_ = np.concatenate((track1, track2), axis=0).min(axis=0)
+        max_ = np.concatenate((track1, track2), axis=0).max(axis=0)
+        subtractor = (max_ + min_) / 2
+        divisor = (max_ - min_) / 2 + 1e-5
+        track1 = (track1 - subtractor) / divisor
+        track2 = (track2 - subtractor) / divisor
+
+        # numpy to torch
+        track1 = torch.tensor(track1, dtype=torch.float).to(self.device)
+        track2 = torch.tensor(track2, dtype=torch.float).to(self.device)
+
+        # unsqueeze channel=1
+        track1 = track1.unsqueeze(0).unsqueeze(0)
+        track2 = track2.unsqueeze(0).unsqueeze(0)
```

Comment:
suggest moving the `transforming numpy to torch` operation before line  240 `confidence = self.model(track_i,`